src/pudl/transform/ferc1.py

```diff
@@ -2672,6 +2672,37 @@ def remove_nones_in_list(list_of_things):
             tbl_meta.loc[xbrl_factoid, "calculations"] = json.dumps(calc_to_update)
         return tbl_meta.reset_index()
 
+    def process_xbrl_metadata_calculations(
+        self,
+    ) -> pd.DataFrame:
+        """Convert xbrl metadata calculations into a table of calculation components."""
+        metadata = self.xbrl_metadata
+        if all(metadata.calculations == "[]"):
+            logger.info(f"{self.table_id.value}: No calculations.")
+            return pd.DataFrame()
+
+        metadata.calculations = metadata.calculations.apply(json.loads)
+        # reset the index post calc explosion so we can merge on index later
+        metadata = metadata.explode("calculations").reset_index(drop=True)
+        calc_comps = (
+            pd.json_normalize(metadata.calculations)
+            .explode("source_tables")
+            .rename(
+                columns={
+                    "name": "xbrl_factoid_calculation_component",
+                    "source_tables": "table_name_calculation_component",
+                }
+            )
+            .merge(
+                metadata.drop(columns=["calculations"]),
+                left_index=True,
+                right_index=True,
+            )
```

Comment:
i didn't know how to keep any of the other `metadata` columns in the `pd.json_normalize` step. so my reaction to that was just to make sure the index was fresssshh above and merge here. it feels okay but if anyone knows how to make `json_normalize` just do the thing lmk